### PR TITLE
Smoothen analysis class docstrings

### PR DESCRIPTION
**Pull request by PicoCentauri**
While @joaomcteixeira and I are working on the MDA command line interface, I found several (possible) improvements to the analysis class docstrings.

Changes made in this Pull Request:

* Added more modules to the `__all__` list
* Changed `: ` -> ` : ` to follow numpydoc docstring guide
* removed default values from docstring
* removed trailing spaces and empty lines

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/encore/clustering/affinityprop.c
Comment (jbarnoud) [inline package/MDAnalysis/analysis/encore/clustering/affinityprop.c:1]: This file is not supposed to be checked into git.

**Review by jbarnoud — COMMENTED** [review]
These are very simple changes, but they are very welcome as well. There is one cython-generated C file that got added in the repo and that should be removed, though.

About the default, we have not been consistent in how we deal with default values in the docstrings. They are places where we write them explicitly between square brackets, places where we do not write them explicitly, and places where we write them using a different format. I think it is worth having a discussion about this (ping @MDAnalysis/coredevs) before we remove them.

I see pros and cons about having them:

* pro: as a user, I like knowing the default value for an argument and sometimes it is unpleasant to have to parse the signature to get it (many arguments,  very long docstring so the signature is far from the description of the parameters, obfuscated defaults, arguments in **kwargs...)
* con: it can easily not match the value in the code

File: package/MDAnalysis/analysis/encore/clustering/affinityprop.c
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/encore/clustering/affinityprop.c]: I removed the file.

**Comment by PicoCentauri** [comment]
Thanks for the comment @jbarnoud. The generated `.c` file is gone.

Let's wait for the reply of the core devs. I removed them because for me it seems that there are more definitions only having the default values in the function or class definitions. The [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#sections) docu are also not very clear how to handle them

> Optional keyword parameters have default values, which are displayed as part of the function signature. They can also be detailed in the description [...]

**Comment by IAlibay** [comment]
> About the default, we have not been consistent in how we deal with default values in the docstrings. They are places where we write them explicitly between square brackets, places where we do not write them explicitly, and places where we write them using a different format. I think it is worth having a discussion about this (ping @MDAnalysis/coredevs) before we remove them.
> 
> I see pros and cons about having them:
> 
> * pro: as a user, I like knowing the default value for an argument and sometimes it is unpleasant to have to parse the signature to get it (many arguments,  very long docstring so the signature is far from the description of the parameters, obfuscated defaults, arguments in **kwargs...)
> * con: it can easily not match the value in the code

I have to say, I'm equally torn between the pro/cons here. I believe there were discussions about this between @orbeckst and @lilyminium in the past (with either the HOLE2 or new density code I think?). 

At the end of the day, as long as we can have uniformity in the docs, it's probably the best outcome. If removing the defaults is the fastest/safest means of ensuring that (which looks like it might be the case), then I'm happy to vote for it.

File: package/MDAnalysis/analysis/hole2/hole.py
Comment (IAlibay) [inline package/MDAnalysis/analysis/hole2/hole.py:1248]: Just a reminder to deal with the PEP8 issues (see pep8speaks)

**Comment by orbeckst** [comment]
If we strip the defaults from the text description then we need to look carefully at any cases where we use `**kwargs` and if necessary make the kwargs explicit (which is better practice anyway).

When the text explains what the default does (e.g., "The default `None` selects all atoms.") then this explanation needs to remain. It's really important for docs to convey the _meaning_ and _intentions_ as this is often not even clear from looking at the code itself.

Overall, I see a slight advantage in clarity and maintainability if we remove "defaults in square brackets" if that's all there is -- this is information that can be kept [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). I just would not cut everything if real information is being lost.

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py:283]: forgot to remove defaults -- the numbers in square brackets!

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py:282]: remov default

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py:285]: remove default [True]

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py
Comment (orbeckst) [reply package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py:282]: There are more instances of missed defaults in the docs here.

File: package/MDAnalysis/analysis/pca.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/pca.py:300]: Rewrite as 
```suggestion
            The number of components to be projected onto. The default value ``None`` maps
```

File: package/MDAnalysis/analysis/pca.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/pca.py:302]: rewrite, see above for example

File: package/MDAnalysis/analysis/pca.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/pca.py:305]: rewrite, see above for example. Also replace n_frames with explicit "the total number of frames in the trajectory".

File: package/MDAnalysis/analysis/pca.py
Comment (orbeckst) [inline package/MDAnalysis/analysis/pca.py:309]: rewrite, if set to ``None`` (the default) then every frame is analyzed (i.e., same as ``step=1``) -- makes things clearer

**Review by orbeckst — CHANGES_REQUESTED** [review]
Generally I'm in favor and thank you for cleaning up!

Found a few missed ones  + suggestions for making things clearer

**Comment by lilyminium** [comment]
I agree with @orbeckst, I am in favour of not repeating default values in the docs.

> pro: as a user, I like knowing the default value for an argument and sometimes it is unpleasant to have to parse the signature to get it (many arguments, very long docstring so the signature is far from the description of the parameters, obfuscated defaults, arguments in **kwargs...)
    con: it can easily not match the value in the code

IMO this con outweighs the pros. 

> if necessary make the kwargs explicit (which is better practice anyway).

And also that in most cases it is much better to just make the kwargs explicit.

File: package/MDAnalysis/analysis/hole2/hole.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/hole2/hole.py:1248]: Thanks. I fixed the lengths

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py]: Done

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py:284]: Done

File: package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py:286]: Done

File: package/MDAnalysis/analysis/pca.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/pca.py]: Done

File: package/MDAnalysis/analysis/pca.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/pca.py:302]: Done

File: package/MDAnalysis/analysis/pca.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/pca.py:305]: Done

File: package/MDAnalysis/analysis/pca.py
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/pca.py:310]: Why didn't we set the default to 1?

Comment (orbeckst) [reply package/MDAnalysis/analysis/pca.py:310]: Because using `None` as step = 1 is the behavior for slicing and range so we keep the user interface "pythonic".

**Review by orbeckst — APPROVED** [review]
There's one open suggestion regarding `step` but either way, this looks good. Thanks @PicoCentauri .

File: package/MDAnalysis/analysis/pca.py
Comment (orbeckst) [reply package/MDAnalysis/analysis/pca.py:310]: @PicoCentauri do you want to improve this doc string, too, or are we squash-merging as is?

**Comment by orbeckst** [comment]
I am going to squash-merge the PR. The failures on Azure are widely varying timings for some of the hypothesis-based tests.

**Comment by orbeckst** [comment]
Thank you @PicoCentauri and @joaomcteixeira !

**Pull request merged**
